eogrow/core/eopatch.py

```diff
@@ -211,7 +212,14 @@ def _eopatch_list_from_id_list(self, id_list: Optional[List[int]]) -> List[str]:
         if id_list is None:
             return list(self.name_to_id_map)
 
-        return [self.name_to_id_map.inverse[eopatch_id] for eopatch_id in id_list]
+        try:
+            return [self.name_to_id_map.inverse[eopatch_id] for eopatch_id in id_list]
+        except KeyError as exception:
+            existing_ids = sorted(self.name_to_id_map.inverse)
+            raise KeyError(
+                f"Given list of patch IDs {large_list_repr(id_list)} is not a sublist of the list of all patch IDs"
+                f" {large_list_repr(existing_ids)}"
```

Comment:
The problem is that both `id_list` and `existing_ids` can have a lot of elements, e.g. even 100K. Printing all of them in an error message would be way too much. In the current implementation we raise an exception from an existing exception which says e.g. `KeyError: 0`, hence a user can know that an id 0 is not one of the existing ids. (Yes, in case of a custom grid users can set any ids they want into the grid file, which was the initial reason for my confusion and preparing this improvement.)

Reply:
ah, the information about the original key error is propagated with `from`.